### PR TITLE
Nuvei integration

### PR DESCRIPTION
**Pull request by mottersheadt**

### CODE REVIEW

File: lib/active_merchant/billing/gateways/nuvei.rb [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+        timestamp = Time.now.utc.strftime("%Y%m%d%H%M%S")
+        add_trans_details(post, money, options, timestamp)
+        add_device_details(post, options)
+        add_payout_details(post, options, timestamp)
+        commit('payout', post, options)
+      end
+
+      private
+
+      # TODO: What is this for Nuvei?
```
Comment (mottersheadt) [inline lib/active_merchant/billing/gateways/nuvei.rb:64]: I have a question out to my Nuvei contact to help with this.

Comment (olkooo) [reply lib/active_merchant/billing/gateways/nuvei.rb:64]: This is the AVS check.
Address Verification Check result which we send to customers.

File: Gemfile [hunk 2]
```diff
@@ -9,4 +9,5 @@ group :test, :remote_test do
   gem 'braintree', '>= 3.0.0', '<= 3.0.1'
   gem 'jwe'
   gem 'mechanize'
+  gem 'rexml'
```
[the remark this answers is not in the document]
Comment (mottersheadt) [reply Gemfile:12]: I needed to have it added or I couldn't run the tests. It is used by a handful of gateways:
image

Are you able to run the tests without it?

File: lib/active_merchant/billing/gateways/nuvei.rb [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+          :merchantSiteId => @merchant_site_id,
+          :timeStamp => timestamp,
+          :checksum => checksum
+        }
+        
+        begin
+          raw_response = ssl_post(url('getSessionToken'), post_data(parameters), request_headers(options))
+          response = parse(raw_response)
+        rescue ResponseError => e
+          e.response.body
```
[the remark this answers is not in the document]
Comment (mottersheadt) [reply lib/active_merchant/billing/gateways/nuvei.rb:64]: It was incorrectly copied from line 175. I will fix.

**Review by berdarm — APPROVED** [review]

**Pull request merged**